pkg/apis/metering/v1/metering_config.go

```diff
@@ -0,0 +1,489 @@
+	TLS                 *TLSSpec                 `json:"tls,omitempty"`
+	Monitoring          *MonitoringSpec          `json:"monitoring,omitempty"`
+	ReportingOperator   *ReportingOperator       `json:"reporting-operator,omitempty"`
+
+	DisableOCPFeatures bool `json:"disableOCPFeatures"`
+	LogHelmTemplate    bool `json:"logHelmTemplate"`
+}
+
+type MeteringConfigSpec struct {
+	// Need to pass this as a pointer to a struct to avoid marshalling the zero-values despite omitempty being specified
+	// See: https://stackoverflow.com/questions/18088294/how-to-not-marshal-an-empty-struct-into-json-with-go
+	Storage             StorageSpec              `json:"storage"`
+	UnsupportedFeatures *UnsupportedFeaturesSpec `json:"unsupportedFeatures,omitempty"`
+	TLS                 *TLSSpec                 `json:"tls,omitempty"`
+	Monitoring          *MonitoringSpec          `json:"monitoring,omitempty"`
+	ReportingOperator   *ReportingOperator       `json:"reporting-operator,omitempty"`
+
+	DisableOCPFeatures bool `json:"disableOCPFeatures,omitempty"`
+	LogHelmTemplate    bool `json:"logHelmTemplate,omitempty"`
+}
+
+type ReportingOperatorSpec struct {
+	Affinity       *Affinity       `json:"affinity,omitempty"`
+	Annotations    *Annotations    `json:"annotations,omitempty"`
+	Config         *Config         `json:"config,omitempty"`
+	Labels         *Labels         `json:"labels,omitempty"`
+	NodeSelector   *NodeSelector   `json:"nodeSelector,omitempty"`
+	Replicas       int             `json:"replicas,omitempty"`
+	Image          *Image          `json:"image,omitempty"`
+	UpdateStrategy *UpdateStrategy `json:"updateStrategy,omitempty"`
+	Resources      *Resources      `json:"resources,omitempty"`
+	Route          *Route          `json:"route,omitempty"`
+	AuthProxy      *AuthProxy      `json:"authProxy,omitempty"`
+}
+type ReportingOperator struct {
+	Spec ReportingOperatorSpec `json:"spec"`
+}
+type Config struct {
+	AllNamespaces       bool                     `json:"allNamespaces,omitempty"`
+	EnableFinalizers    bool                     `json:"enableFinalizers,omitempty"`
+	LeaderLeaseDuration string                   `json:"leaderLeaseDuration,omitempty"`
+	LogDDLQueries       bool                     `json:"logDDLQueries,omitempty"`
+	LogDMLQueries       bool                     `json:"logDMLQueries,omitempty"`
+	LogReports          bool                     `json:"logReports,omitempty"`
+	LogLevel            string                   `json:"logLevel,omitempty"`
+	Aws                 *AWS                     `json:"aws,omitempty"`
+	Hive                *ReportingOperatorHive   `json:"hive,omitempty"`
+	Presto              *ReportingOperatorPresto `json:"presto,omitempty"`
+	Prometheus          *Prometheus              `json:"prometheus,omitempty"`
+	TLS                 *APITLS                  `json:"tls,omitempty"`
+}
+
+type Affinity struct {
+}
+type Annotations struct {
```

Comment:
https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta shows the types of annotations/labels which are `map[string]string`